src/main/java/com/bigludo/ClientHandler.java

```diff
@@ -1,22 +1,33 @@
 package com.bigludo;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
 import java.net.Socket;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class ClientHandler implements IClient{
-
+public class ClientHandler implements IClient, Runnable{
     private static final Logger log = LogManager.getLogger(ClientHandler.class);
-
     private String ID;
     private Socket clientSocket;
-    private IServer server;    
+    private IServer server;
+    private BufferedReader br;
+    private Thread clientThread;
 
     public ClientHandler (String id, Socket cs, IServer srv) {
         ID = id;
         clientSocket = cs;
         server = srv;
         log.info("ClientHandler created " + id);
+
+        try {
+            clientThread = new Thread(this);
```

Comment:
Jag skulle flytta starten av tråden till efter att du skapat BufferReader. Tråden kommer direkt försöka lägga sig och lyssna på inputstream och man skulle kunna få något race-condition.

Reply:
Dvs. kör rad 27 (br = new BufferReader...) före rad 25 och 26 (clientThread = new Thread(this); och clientThread.start();). Det viktigaste är väl egentligen clientThread.start();